DisguiseUnityRenderStream/Runtime/PipelineAbstraction/DepthCopy.cs

```diff
@@ -0,0 +1,193 @@
+        /// </summary>
+        public enum Mode
+        {
+            Raw,
+            Eye,
+            Linear01
+        }
+        
+        /// <summary>
+        /// Parameters for the depth copy.
+        /// </summary>
+        public struct FrameData
+        {
+            /// <summary>
+            /// The texture to which the depth will be written to.
+            /// </summary>
+            public RenderTexture m_depthOutput;
+
+            public bool IsValid => m_depthOutput != null;
+        }
+        
+        // TODO: use ShaderVariantCollection + preload in GraphicsSettings
+        // Once that's done DepthCopy can be made Serializable
+        struct ShaderVariantResources : IDisposable
+        {
+            public ShaderResources m_Raw;
+            public ShaderResources m_Eye;
+            public ShaderResources m_linear01;
+
+            public bool IsLoaded => m_Raw.IsLoaded && m_Eye.IsLoaded && m_linear01.IsLoaded;
+
+            public void Create(string shaderPath, string shaderPass)
+            {
+                m_Raw.Create(shaderPath, shaderPass);
+                m_Eye.Create(shaderPath, shaderPass);
+                m_linear01.Create(shaderPath, shaderPass);
+                
+                m_Raw.SetKeyword("DEPTH_COPY_RAW", true);
+                m_Eye.SetKeyword("DEPTH_COPY_EYE", true);
+                m_linear01.SetKeyword("DEPTH_COPY_LINEAR01", true);
+            }
+
+            public void Dispose()
+            {
+                m_Raw.Dispose();
+                m_Eye.Dispose();
+                m_linear01.Dispose();
+            }
+        }
+        
+        struct ShaderResources : IDisposable
+        {
+            public Shader m_shader;
+            public Material m_material;
+            public int m_pass;
+
+            public bool IsLoaded => m_shader != null && m_material != null && m_pass >= 0;
+
+            public void SetKeyword(string keyword, bool value)
+            {
+                if (IsLoaded)
+                {
+                    if (value)
+                        m_material.EnableKeyword(keyword);
+                    else
+                        m_material.DisableKeyword(keyword);
+                }
+            }
+
+            public void Create(string shaderPath, string shaderPass)
+            {
+                m_shader = Shader.Find(shaderPath);
+                if (m_shader != null)
+                {
+                    m_material = CoreUtils.CreateEngineMaterial(m_shader);
+                    m_pass = m_material.FindPass(shaderPass);
+                }
+            }
+
+            public void Dispose()
+            {
+                CoreUtils.Destroy(m_material);
+            }
+        }
+
+        public Mode mode
+        {
+            get => m_mode;
+            set
+            {
+                m_mode = value;
+                
+                switch (m_mode)
```

Comment:
Prefer the new switch expression syntax (where it makes sense). It's more compact and more readable.
```csharp
m_shaderResources = m_mode switch {
  Mode.Raw => s_shaderVariantResources.m_Raw,
  Mode.Eye => s_shaderVariantResources.m_Eye,
  Mode.Linear01 => s_shaderVariantResources.m_linear01,
  _ => throw new ArgumentOutOfRangeException()
};
```